src/dao/user.dao.ts

```diff
@@ -7,32 +7,60 @@ import { formSchema } from "@/shared/onboarding";
 
 class UserDAO {
   async createUser(loginUser: NextUser) {
-    const existingUser = await this.fetchUserBy({
-      email: loginUser.email ?? raise("E-mail not found"),
-    });
+    if (!loginUser.email) {
+      throw raise("E-mail not found");
+    }
+    const existingUser = await this.fetchUserBy({ email: loginUser.email });
```

Comment:
I suggest we revert this change and try to understand the current implementation

THis is being used on the App UI